tsconfig.json

```diff
@@ -6,7 +6,7 @@
     "moduleResolution": "node",
     "target": "es6",
     "outDir": "./lib",
-    "lib": ["es6", "dom"],
+    "lib": ["es6", "dom", "esnext.array"],
```

Comment:
What does this extra lib in tsconfig.json get us?

Reply:
According to [this thread](https://stackoverflow.com/questions/53556409/typescript-flatmap-flat-flatten-doesnt-exist-on-type-any), the [`flatMap` function](https://github.com/Microsoft/TypeScript/blob/ef7997cb73ce46081cbfebc0edd2a0295ba30c33/lib/lib.esnext.array.d.ts).

Without this, TypeScript complains that `Property 'flatMap' does not exist on type 'ISyntaxNode[]'`.

Reply:
Got it! Possibly stupid question: does this mean this code won't work for browsers that don't support `flatMap` on array (e.g., Chrome pre-69, Firefox pre-62 according to [this MDN page](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap))? Or will `flatMap` get cross-browser compatible JavaScript by the TypeScript compiler?

If this breaks compatibility with older browsers, we probably want to consider an alternate implementation that doesn't use the flatMap method.

Reply:
[You're right](https://stackoverflow.com/questions/39837678/why-no-array-prototype-flatmap-in-javascript#comment99750916_56586455), I forgot this needs to run in a browser. I don't know much about browser compatibility, would [this](https://github.com/you-dont-need/You-Dont-Need-Lodash-Underscore#_flattendeep) implementation be sufficiently compatible?

Reply:
Maybe we don't even need flatMap? Want to check out the suggested fix in data-flow.ts?

Reply:
Yep, I removed the extra lib from the tsconfig!